src/calendar/CalendarUtils.js

```diff
@@ -36,26 +36,45 @@ export function generateUid(event: CalendarEvent, timestamp: number): string {
 }
 
 /**
- * Accepts 2, 2:30, 2:5, 02:05, 02:30, 24:30
+ * Accepts 2, 2:30, 2:5, 02:05, 02:30, 24:30, 2430, 12:30pm, 12:30 p.m.
  */
 export function parseTime(timeString: string): ?{hours: number, minutes: number} {
-	const suffix = timeString.substr(-2).toUpperCase()
-	if (suffix === "AM" || suffix === "PM") {
-		timeString = timeString.slice(0, -2)
-	}
-	const parts = timeString.split(":")
-	let hours = filterInt(parts[0])
-	let minutes = 0
-	if (parts.length === 2) {
-		minutes = filterInt(parts[1])
+	var suffix = null
+	var hours = NaN
+	var minutes = NaN
```

Comment:
Please no `var`, only `let`/`const`.
Also you don't have to initialize `hours`/`minutes` I think. If there's a branch they're not assigned it, it should be an error.